heracles/dices/shrinkage.py

```diff
@@ -160,19 +161,35 @@ def _gaussian_covariance(cls, key):
         cov: covariance matrix
     """
     a1, b1, a2, b2, i1, j1, i2, j2 = key
-    clkey1 = format_key((a1, a2, i1, i2))
-    clkey2 = format_key((b1, b2, j1, j2))
-    clkey3 = format_key((a1, b2, i1, j2))
-    clkey4 = format_key((b1, a2, j1, i2))
-    cl1 = cls[clkey1].array
-    cl2 = cls[clkey2].array
-    cl3 = cls[clkey3].array
-    cl4 = cls[clkey4].array
-    # Compute the Gaussian covariance
+    cl1 = _get_cl((a1, a2, i1, i2), cls)
+    cl2 = _get_cl((b1, b2, j1, j2), cls)
+    cl3 = _get_cl((a1, b2, i1, j2), cls)
+    cl4 = _get_cl((b1, a2, j1, i2), cls)
     cov = cl1 * cl2 + cl3 * cl4
     return cov
 
 
+def _get_cl(key, cls):
+    """
+    Internal method to get a Cl from a dictionary of Cls.
+    Check if the key exists if not tries to find the symmetric key.
+    input:
+        key: key of the Cl
+        cls: dictionary of Cls
+    returns:
+        cl: Cl
+    """
+    if key in cls:
+        return cls[key].array
+    else:
+        a, b, i, j = key
+        key_sym = (b, a, j, i)
+        if key_sym in cls:
+            return cls[key_sym].array
```

Comment:
This doesn't do all of the `EB <-> BE` flips if the array has multiple spin-weighted fields.